modules/sbt-plugin/src/main/scala/ch/epfl/scala/debugadapter/sbtplugin/DebugAdapterPlugin.scala

```diff
@@ -104,13 +120,59 @@ object DebugAdapterPlugin extends sbt.AutoPlugin {
     startMainClassDebugSession := mainClassSessionTask.evaluated,
     startRemoteDebugSession := remoteSessionTask.evaluated,
     stopDebugSession := stopSessionTask.value,
+    debugAdapterClassesObserver := PublishSubject.create(),
     Keys.compile / Keys.javacOptions := {
       val jo = (Keys.compile / Keys.javacOptions).value
       if (jo.exists(_.startsWith("-g"))) jo
       else jo :+ "-g"
+    },
+    Keys.compile := {
+      val currentAnalysis: CompileAnalysis = Keys.compile.value
+      val previousAnalysis = Keys.previousCompile.value.analysis
+      val observer = debugAdapterClassesObserver.value
+
+      observer.onNext {
+        getNewerClassFiles(currentAnalysis.readStamps, previousAnalysis, fileConverter.value, classDirectory.value)
```

Comment:
I prefer to use `Keys.foo.value` rather than `foo.value` with `import Keys.{foo}`.

Also, it is good to have all `*.value` at the beginning to see the dependencies between tasks.
```suggestion
        val fileConverter = Keys.fileConverter.value
        val classDirectory = Keys.classDirectory.value
        
        getNewerClassFiles(currentAnalysis.readStamps, previousAnalysis, fileConverter, classDirectory)
```

Reply:
Also I think we should treat the case of an empty previousAnalysis here:

```scala
if  (previousStamps.isPresent()) {
  val classesToUpdate = getNewerClassFiles(currentAnalysis.readStamps, previousAnalysis.get.readStamps, fileConverter, classDirectory)
}
```

This would simplify the implementation of `getNewerClassFiles`.